mrmustard/lab_dev/states/base.py

```diff
@@ -935,13 +938,15 @@ def __rshift__(self, other: CircuitComponent) -> CircuitComponent:
         not needed and the method returns a new ``Ket``.
 
         Returns a ``DM`` or a ``Ket`` when the wires of the resulting components are compatible
-        with those of a ``DM`` or of a ``Ket``. Returns a ``CircuitComponent`` otherwise.
+        with those of a ``DM`` or of a ``Ket``. Returns a ``CircuitComponent`` otherwise and a scalar if there are no wires left.
```

Comment:
```suggestion
        with those of a ``DM`` or of a ``Ket``. Returns a ``CircuitComponent`` otherwise and a (batched) scalar if there are no wires left.
```